.github/workflows/nm-benchmark.yml

```diff
@@ -108,9 +121,81 @@ jobs:
           venv: TEST
 
       - name: store benchmark result artifacts
+        if: success()
         uses: actions/upload-artifact@v4
-        if: success() || failure()
         with:
           name: ${{ github.run_id }}-${{ inputs.label }}
           path: benchmark-results
           retention-days: 10
+
+      # Produce GHA benchmark JSONs
+      - name: make github-action-benchmark JSONs 
+        uses: ./.github/actions/nm-produce-gha-benchmark-json
+        with:
+          vllm_benchmark_jsons_path: benchmark-results 
+          # Metrics that are "better" when the value is greater are stored here
+          bigger_is_better_output_file_path: gh-action-benchmark-jsons/bigger_is_better.json
+          # Metrics that are "better" when the value is smaller are stored here
+          smaller_is_better_output_file_path: gh-action-benchmark-jsons/smaller_is_better.json
+          python: ${{ inputs.python }}
+          venv: TEST
+
+      - name: set gh action benchmark input artifact name
+        id: set_gh_action_benchmark_input_artifact_name
+        run: |
+          GH_ACTION_BENCHMARK_INPUT_ARTIFACT_NAME=`echo "gh_action_benchmark_jsons-${{ github.run_id }}-${{ inputs.label }}"`
+          echo "gh_action_benchmark_input_artifact_name=$GH_ACTION_BENCHMARK_INPUT_ARTIFACT_NAME" >> $GITHUB_OUTPUT
+
+      - name: store gh action benchmark input artifacts
+        if: success()
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ steps.set_gh_action_benchmark_input_artifact_name.outputs.gh_action_benchmark_input_artifact_name}}
+          path: gh-action-benchmark-jsons 
+          retention-days: 1
+
+  NM_GH_ACTION_BENCHMARK:
+
+    needs: BENCHMARK 
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    permissions:
+      # Permissions required to be able to push to the nm-gh-pages branch 
+      contents: write
```

Comment:
Calling out this permission - This is required for the action to be able to write to `nm-gh-pages` branch. However, it grants write access to the entire repo and not just that branch. I couldn't find a finer grained permission.

Reply:
@robertgshaw2-neuralmagic

Reply:
Resolution : Lets keep it for now. We need to audit this permission at the company level. 
If this is not okay there is another option : 
   There is an option to point the action to a different repo - maybe we create one and have the action do all the write to that repo.